### ISSUE TITLE
different logits

### ISSUE THREAD

**Issue by speeding-motor** [LoveHRTF/YouOnlyLookOnce-TF2.0#12]
there have a few different logit in you yolo code with normal, such as 
1、model：you always use the padding='same', but it's wrong 
2、loss: when you compute the loss, you sum the all of the example / batch_size to get the batch average loss and return one loss_value , but it need to return loss with shape [1, batch_size]

**Comment by LoveHRTF** [comment]
The original [You Only Look Once](https://arxiv.org/pdf/1506.02640.pdf) paper did not specify which type of padding method to use. Although YoloV3 uses constant padding of 1, it would be totally valid for using same padding for the sake of preserving specified layer size. Could you specify which part goes wrong so I can fix it.

Also, although this is a little bit fundamental, but the whole point of using SGD was to averaging the image loss to a batch loss for a smooth weight update process. I'm not seeing any reason for returning loss for each individual image, otherwise please let me know.

**Comment by speeding-motor** [comment]
yeah you are right, the point to use batch gradient was to average the loss of the batch example ,but maybe it need to compute the gradient of the each of sample then average the gradient

**Comment by LoveHRTF** [comment]
These operations were vectorized. All the summations and averaging was done in loss functions. Please refer to line 118 to 165 in model.py.

**Comment by speeding-motor** [comment]
yeah, you are right, even we don't mean the loss , the Tensorflow frame will do that for us,

**Comment by speeding-motor** [comment]
when I run my code ,the finally loss is about 60.0, but the gradient is lost in the last two layers(45, total layer is 47), have you ever met the problem...?

**Comment by LoveHRTF** [comment]
no
